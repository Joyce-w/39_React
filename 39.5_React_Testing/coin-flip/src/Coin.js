import React from "react"
import "./Coin.css"

const Coin = ({data}) => {
    return (
        <div className="Coin">
            <img className="Coin-img" src={data.image} alt={data.name}></img>
        </div>
    
    )
}

export default Coin;