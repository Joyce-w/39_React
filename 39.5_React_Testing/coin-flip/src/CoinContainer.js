import React, {useState} from "react"
import Coin from "./Coin"
import heads from "./dime_heads.jpg"
import tails from "./dime_tails.jpg"
import "./CoinContainer.css"

const data = [
    {
    id: 1,
    image: heads,
    name: "dime_head"
    },
    {
    id: 2,
    image: tails,
    name: "tail_head"
    }
]

const CoinContainer = () => {
    //create random num to flip
    let rand = () => {
        const idx = Math.floor(Math.random() * 2);
        return idx
    } 
    //state for random coin flip
    const [currSide, flip] = useState(0);

    //total counter
    const [total, add] = useState(0);
    //head counter
    const[headCount, addHead] = useState(0)
    //tail counter
    const [tailCount, addTail] = useState(0)

    const flipCoin = () => {
        flip(rand())
        add(total + 1)

        if (data[currSide].name === "dime_head") {
        addHead(headCount + 1)
        }
        else {
            addTail(tailCount + 1)
        }
    }
    
    return (
        <div className="CoinContainer">
            <h1>Flip a coin </h1>
            <div>
                <Coin data={ data[currSide] }/>                
            </div>
            <p>Out of {total} flips, there have been {headCount} heads and { tailCount } tails. </p>
            <button onClick={() => flipCoin()}>Flip Coin</button>
        </div>
    )
}

export default CoinContainer;