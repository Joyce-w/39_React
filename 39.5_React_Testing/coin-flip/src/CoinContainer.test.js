import React from "react"
import { render } from "@testing-library/react"
import CoinContainer from "./CoinContainer"
import Coin from "./Coin"

//smoke test
it("renders w/o crashing", () => {
    render(<CoinContainer/>)
})

//snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<CoinContainer />);
    expect(asFragment()).toMatchSnapshot();
})

test('if button exists', () => {
    const { getByText } = render(<CoinContainer />)
    const buttonText = getByText("Flip Coin");
    expect(buttonText).toBeInTheDocument();

})