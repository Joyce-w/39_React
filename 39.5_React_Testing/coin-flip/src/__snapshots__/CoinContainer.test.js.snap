// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CoinContainer"
  >
    <h1>
      Flip a coin 
    </h1>
    <div>
      <div
        class="Coin"
      >
        <img
          alt="dime_head"
          class="Coin-img"
          src="dime_heads.jpg"
        />
      </div>
    </div>
    <p>
      Out of 0 flips, there have been 0 heads and 0 tails. 
    </p>
    <button>
      Flip Coin
    </button>
  </div>
</DocumentFragment>
`;
